src/Lykke.Common.ApiLibrary/Middleware/RequestUtils.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System;
+using System.IO;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+
+namespace Lykke.Common.ApiLibrary.Middleware
+{
+    internal static class RequestUtils
+    {
+        public static async Task<string> GetRequestBodyAsync(HttpContext context)
+        {
+            // request body might be already read at the moment 
+            if (context.Request.Body.CanSeek)
+            {
+                context.Request.Body.Seek(0, SeekOrigin.Begin);
+            }
+
+            using (var stream = new MemoryStream())
```

Comment:
Should we create a copy of body only to truncate output? ReadAsync(bodyPart, 0, len); len is a maximum, we can pass there simple maxBodySize

Reply:
Fixed